src/openms/include/OpenMS/FORMAT/FileHandler.h

```diff
@@ -166,76 +205,222 @@ namespace OpenMS
 
       @param filename The name of the file to store the data in.
       @param exp The experiment to store.
+      @param allowed_types a FileTypeList containing the types of files to support. If not empty the extension, and failing that the contents are checked to make sure they match
       @param log Progress logging mode
 
       @exception Exception::UnableToCreateFile is thrown if the file could not be written
     */
-    void storeExperiment(const String& filename, const MSExperiment& exp, ProgressLogger::LogType log = ProgressLogger::NONE);
+    bool storeExperiment(const String& filename, const PeakMap& exp, const std::vector<FileTypes::Type> allowed_types = std::vector<FileTypes::Type>(), ProgressLogger::LogType log = ProgressLogger::NONE);
 
     /**
       @brief Loads a file into a FeatureMap
 
       @param filename the file name of the file to load.
       @param map The FeatureMap to load the data into.
-      @param force_type Forces to load the file with that file type. If no type is forced, it is determined from the extension (or from the content if that fails).
 
       @return true if the file could be loaded, false otherwise
```

Comment:
do you know when false is returned?
it sounds to me that it throws exceptions.
seems kind of weird mixing exceptions and returning a bool

Reply:
agree.
Providing users a way to ask if we can load/store a file type and then throw exceptions during loading if she uses it in an unsupported way.